desk/app/channels.hoon

```diff
@@ -622,14 +782,22 @@
       %kick
     ?:  (~(has by voc) nest plan)
       cor  :: subscription ended politely
-    (give %kick ~[path v0+path v1+path] ~)
+    (give %kick ~[path v0+path v1+path v2+path] ~)
   ::
       %fact
-    =.  cor  (give %fact ~[path v0+path v1+path] cage.sign)
-    =.  cor  (give %kick ~[path v0+path v1+path] ~)
+    =.  cor  (give %fact ~[v2+path] cage.sign)
+    ::  send legacy said:v7 response
+    ::
+    =.  cor  %^  give  %fact
+        ~[path v0+path v1+path]
+      ?.  ?=(%channel-said-2 p.cage.sign)
+        cage.sign
+      =+  !<(=said:c q.cage.sign)
+      channel-said+!>((to-said-1:utils said))
```

Comment:
I would make sure to specify the different cases that we handle here. Optionally, you can consider unifying this with the `?+` below.

```suggestion
    =.  cor
      %^  give  %fact
        ~[path v0+path v1+path]
      ?+  p.cage.sign  ~|(funny-mark+p.cage.sign !!)
        %channel-denied  cage.sign
      ::
          %channel-said-2
        =+  !<(=said:c q.cage.sign)
        channel-said+!>((to-said-1:utils said))
      ==
```